internal/kafka/internal/services/quota/ams_quota_service.go

```diff
@@ -396,3 +398,79 @@ func (q amsQuotaService) GetSubscriptionByID(subscriptionID string) (*amsv1.Subs
 
 	return resp.Body(), true, nil
 }
+
+// Checks if an organisation has a SKU entitlement and that entitlement is still active in AMS
+func (q amsQuotaService) IsQuotaEntitlementActive(kafka *dbapi.KafkaRequest) (bool, error) {
+	kafkaBillingModel, err := q.kafkaConfig.GetBillingModelByID(kafka.InstanceType, kafka.ActualKafkaBillingModel)
+	if err != nil {
+		return false, err
+	}
+
+	amsRelatedResourceBillingModel, err := q.getAMSRelatedResourceBillingModel(kafka.SubscriptionId, kafkaBillingModel)
+	if err != nil || arrays.StringEmptyPredicate(amsRelatedResourceBillingModel) {
+		return false, err
+	}
+
+	orgID, err := q.amsClient.GetOrganisationIdFromExternalId(kafka.OrganisationId)
+	if err != nil {
+		return false, err
+	}
+
+	quotaCostRelatedResourceFilter := ocm.QuotaCostRelatedResourceFilter{
+		ResourceName: &kafkaBillingModel.AMSResource,
+		Product:      &kafkaBillingModel.AMSProduct,
+		BillingModel: &amsRelatedResourceBillingModel,
+	}
+	quotaCosts, err := q.amsClient.GetQuotaCosts(orgID, true, false, quotaCostRelatedResourceFilter)
+	if err != nil {
+		return false, err
+	}
+
+	// SKU not entitled to the organisation
+	if len(quotaCosts) == 0 {
+		glog.V(10).Infof("ams quota cost not found for Kafka instance %q in organisation %q: %+v", kafka.ID, orgID, quotaCostRelatedResourceFilter)
+		return false, nil
+	}
+
+	if len(quotaCosts) > 1 {
+		glog.Warning("more than 1 quota cost was returned with the following related resource filter %+v for organisation %q", orgID, quotaCostRelatedResourceFilter)
```

Comment:
Why is this a warning? I think in that case we should return an error as it is unexpected and we don't know what numbers we would be comparing